runtime/compiler/runtime/IProfiler.cpp

```diff
@@ -4853,67 +4901,309 @@ void TR_IProfiler::dumpIPBCDataCallGraph(J9VMThread* vmThread)
       fprintf(stderr, "Cannot allocate memory. Bailing out.\n");
       return;
       }
+   traverseIProfilerTableAndCollectEntries(&aggregationHT, vmThread, true/*collectOnlyCallGraphEntries*/);
+   aggregationHT.sortByNameAndPrint();
 
-   J9JavaVM *javaVM = vmThread->javaVM;
-   J9InternalVMFunctions *vmFunctions = javaVM->internalVMFunctions;
-   TR_J9VMBase * fe = TR_J9VMBase::get(javaVM->jitConfig, vmThread);
+   fprintf(stderr, "Finished dumping info\n");
+   }
 
-   TR::VMAccessCriticalSection dumpCallGraph(fe);
+uintptr_t
+TR_IProfiler::createBalancedBST(TR_IPBytecodeHashTableEntry **ipEntries, int32_t low, int32_t high, uintptr_t memChunk, TR_J9SharedCache *sharedCache)
```

Comment:
This seems to be almost equivalent to 
```
uintptr_t
TR_IProfiler::createBalancedBST(uintptr_t *pcEntries, int32_t low, int32_t high, uintptr_t memChunk,
                                TR_J9SharedCache *sharedCache)
```
except for the fact that it does not call `profilingSample` to get the `entry`. Is there any way refactor the code to prevent this duplication?

Reply:
The two functions have different parameters and are recursive. The closest thing would be a template solution, something like this:
```